### PR TITLE
Add new triggerElement prop

### LINKED ISSUE #38: auro-interruption: return user to the trigger element upon modal closure when in a nested shadow root
## Is your feature request related to a problem? Please describe.

It's best practice when closing modals for the focus to be put back on the element the modal was triggered from. Depending on implementation, this can be somewhat tricky. 

## Describe the solution you'd like

A new attribute added onto auro-interruption 'triggerRef' could take either an id or a query selector to return the user to the element that triggered the modal. In some cases like for upsells that are time-launched and not by a trigger by the guest, this feature would be extra helpful.

## Describe alternatives you've considered

There's a lot of ways to implement this in consuming codebases, but this feels like a developer experience improvement we can address from the design system.

### PR DESCRIPTION
**Pull request by geoffrich**
# Alaska Airlines Pull Request

Fixes #38

## Summary:

This PR adds a new `triggerElement` prop so the consumer can indicate which element should be focused when the modal closes. In most cases, this is not needed and the component will use `document.activeElement` by default. However, when the trigger element is inside another shadow root, `document.activeElement` will be equal to the parent custom element, not the trigger. In this case, the user of the component should set `triggerElement` to the desired element to preserve keyboard accessibility. See the [example element](https://github.com/AlaskaAirlines/auro-interruption/blob/550dac097d2f8c3cd81c1b2f86bb8fd72eab807b/test/nested-shadow.js) from the unit tests for an example.

I'll be creating a follow-up story so that the component can recursively search the shadow roots of the active element and correctly determine the original trigger, making this property less necessary. However, exposing this property is still a useful escape hatch.

This PR also includes code style updates to make ESLint happy. It was only looking at the `auro-*.js` elements, and running it on the other JS files exposed some other linting errors. 

## Type of change:

Please delete options that are not relevant.

- [x] New capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -63,7 +64,7 @@ module.exports = {
       "id-length": "error",
       "id-match": "error",
       "implicit-arrow-linebreak": "error",
-      "indent": ["error", 2],
+      "indent": ["error", 2, { "ignoredNodes": ["TemplateLiteral > *"] }],
```
Comment (blackfalcon) [inline .eslintrc.js:67]: This repo is pre-v.3.5 so this update is correct. But I will also add this to the esLint repo. 

https://github.com/AlaskaAirlines/eslint-config-auro/issues/4

File: .eslintrc.js [hunk 2]
```diff
@@ -127,7 +128,7 @@ module.exports = {
       "no-lone-blocks": "error",
       "no-lonely-if": "error",
       "no-loop-func": "error",
-      "no-magic-numbers": "error",
+      "no-magic-numbers": ["error", { "ignore": [0, 1] }],
```
Comment (blackfalcon) [inline .eslintrc.js:131]: I like this! I hate this rule. 👏

File: src/component-base.js [hunk 3]
```diff
@@ -86,7 +89,11 @@ export default class ComponentBase extends LitElement {
 
     this._open = val;
     this.requestUpdate('open', oldVal).then(() => {
-      this.open ? this.openDialog() : this.closeDialog()
+      if (this.open) {
```
Comment (blackfalcon) [inline src/component-base.js:92]: Why remove the ternary?

**Review by blackfalcon — APPROVED** [review]
This is a great fix to the issue and some healthy clean up too. 

Only one question as to change of `if()` format.

File: src/component-base.js (hunk 3 above)
Comment (geoffrich) [reply src/component-base.js:92]: I believe eslint didn’t like the result of the ternary not being assigned to something.

**Pull request merged**